cocos/misc/lodgroup-component.ts

```diff
@@ -0,0 +1,605 @@
+    set screenUsagePercentage (val: number) {
+        this._screenUsagePercentage = val;
+        this._LODData.screenUsagePercentage = val;
+    }
+
+    @type([MeshRenderer])
+    get renderers (): readonly MeshRenderer[]  {
+        return this._renderers;
+    }
+
+    /**
+     * @en reset _renderers to meshList or [], LODData's model will be reset too.
+     * @zh 重置 _renderers 为 meshList或空数组, LODData上的model也会被重置
+     */
+    set renderers (meshList: readonly MeshRenderer[]) {
+        this._renderers.length = 0;
+        this._LODData.models.length = 0;
+        for (let i = 0; i < meshList.length; i++) {
+            this._renderers[i] = meshList[i];
+            const model = meshList[i]?.model;
+            if (model) {
+                this._LODData.models.splice(0, 0, model);
+            }
+        }
+    }
+
+    /**
+     * @engineInternal
+     * @en Get the total number of all mesh's triangle.
+     * @zh 获取所有模型的三角形总数
+     */
+    @editable
+    @type([Number])
+    get trianglesCount () : number[] {
+        const tris: number[] = [];
+        this._renderers.forEach((meshRenderer: MeshRenderer | null) => {
+            let count = 0;
+            if (meshRenderer && meshRenderer.mesh) {
+                const primitives = meshRenderer.mesh.struct.primitives;
+                primitives?.forEach((subMesh: Mesh.ISubMesh) => {
+                    if (subMesh && subMesh.indexView) {
+                        count += subMesh.indexView.count;
+                    }
+                });
+            }
+            tris.push(count / 3);
+        });
+        return tris;
+    }
+
+    /**
+     * @en Get the number of LOD.
+     * @zh 获取LOD的数量
+     */
+    get rendererCount () : number { return this._renderers.length; }
+
+    /**
+      * @engineInternal
+      * @en Get internal LOD object.
+      */
+    get lodData () { return this._LODData; }
+
+    /**
+     * @en Insert a [[MeshRenderer]] before specific index position.
+     * @zh 在指定的数组索引处插入一个[[MeshRenderer]]
+     * @param index, the rendering array is indexed from 0. If - 1 is passed, it will be added to the end of the list.
+     * @param renderer the mesh-renderer object
+     * @returns The renderer inserted
+     */
+    insertRenderer (index: number, renderer: MeshRenderer): MeshRenderer {
+        // make sure insert at the tail of the list.
+        if (index < 0 || index > this._renderers.length) {
+            index = this._renderers.length;
+        }
+        this._renderers.splice(index, 0, renderer);
+        if (renderer.model) {
+            this._LODData.models.splice(0, 0, renderer.model);
+        }
+        return renderer;
+    }
+
+    /**
+     * @en Delete the [[MeshRenderer]] at specific index position.
+     * @zh 删除指定索引处的[[MeshRenderer]]
+     * @param index 0 indexed position in renderer array, when -1 is specified, the last element will be deleted
+     * @returns The renderer deleted
```

Comment:
The same as above. And the return value could be null, should mention it in doc.